pool/pool.go

```diff
@@ -1849,9 +1849,11 @@ func (p *innerPool) connection() (client, error) {
 	return nil, errors.New("no healthy client")
 }
 
-func formCacheKey(address string, key *ecdsa.PrivateKey) string {
-	k := keys.PrivateKey{PrivateKey: *key}
-	return address + k.String()
+func formCacheKey(address string, signer neofscrypto.Signer) string {
+	b := make([]byte, signer.Public().MaxEncodedSize())
+	signer.Public().Encode(b)
```

Comment:
We need a separate commit for this private->public change.

Reply:
i'm not sure it is possible to do atomic. This change had been made because the function signature was changed after replacing PrivateKey to Signer. Extracting this in separate commit leads not compiled code in previous commit

Reply:
You can make it before changing to `Signer`.